xbmc/favourites/FavouritesUtils.h

```diff
@@ -21,4 +23,8 @@ bool MoveItem(CFileItemList& items, const std::shared_ptr<CFileItem>& item, int
 bool RemoveItem(CFileItemList& items, const std::shared_ptr<CFileItem>& item);
 bool ShouldEnableMoveItems();
 
+bool ExecuteAction(const std::string& execString);
```

Comment:
it'd be great if this somehow could be refactored later so that a "single source of truth" could exist. Conversion to expected argument type should probably be done by the caller.
std::variant could be another alternative but in this case (imoh) only hides the design flaws

Reply:
Sorry, what exactly do you mean?

> Conversion to expected argument type should probably be done by the caller.

I tend to disagree. Do you have the same argument against ctors taking arguments of different types?

> only hides the design flaws

 Which ones exactly?

Reply:
I mean that `ExecuteAction` as the name indicates should receive a given action of a **concrete type** as the argument so we have a clean interface for those using the API. 
The way it's done here works for generic strings, CExecStrings, CFavouritesURL and so on. Someone not familiar with the kodi codebase will find this "freedom" confusing. You delegate this degree of flexibility to the consumer interface instead of having a concrete object/class that encapsulates it. The comparison with ctors with arguments of different types is imoh somehow different as it's linked to object creation rather than object usage. Personally I'd prefer to have something like a `CFavouriteAction` class that could be constructed from strings, urls and so on while having a single service interface `ExecuteAction` that only executes the action and only receives a `CFavouriteAction`. We've something similar with `CFileItems` (although not the best example ever since it has grown massively). You can construct an item using different arguments but you don't have multiple service methods for the same argument types you use to construct a fileitem...
Not asking you to do this, just pointing out my opinion.

Reply:
Thanks. Let me think about this.